main.tf

```diff
@@ -0,0 +1,166 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.16"
+    }
+  }
+
+  required_version = ">= 1.5.7"
+
+  backend "s3" {
+    bucket               = "ccao-terraform-state-us-east-1"
+    key                  = "workspaces/default/terraform.tfstate"
+    region               = "us-east-1"
+    workspace_key_prefix = "workspaces"
+  }
+}
+
+provider "aws" {
+  region = "us-east-1"
+}
+
+locals {
+  s3_prefix                = "scripts/sales-val"
+  s3_bucket_data_warehouse = terraform.workspace == "prod" ? "ccao-data-warehouse-us-east-1" : aws_s3_bucket.data_warehouse[0].id
+  s3_bucket_glue_assets    = terraform.workspace == "prod" ? "ccao-glue-assets-us-east-1" : aws_s3_bucket.glue_assets[0].id
+  glue_job_name            = "sales_val_flagging${terraform.workspace == "prod" ? "" : "_${terraform.workspace}"}"
+  glue_crawler_name        = "ccao-data-warehouse-sale-crawler${terraform.workspace == "prod" ? "" : "-${terraform.workspace}"}"
+}
+
+variable "iam_role_arn" {
+  type        = string
+  description = "IAM role ARN to use to run the job"
+  sensitive   = true
+  nullable    = false
+}
+
+variable "commit_sha" {
+  type        = string
+  description = <<EOF
+    "Hash of the current git commit, to use for versioning the model. "
+    "Generate with `git rev-parse --short HEAD`"
+  EOF
+  nullable    = false
+}
+
+resource "aws_s3_bucket" "glue_assets" {
+  # Prod buckets are managed outside this config
+  count         = terraform.workspace == "prod" ? 0 : 1
```

Comment:
It is perhaps [a miserable hack](https://tomharrisonjr.com/terraform-count-is-a-miserable-hack-d58a6ffbf422), but setting the [`count`](https://developer.hashicorp.com/terraform/language/meta-arguments/count) of a resource to `0` on specific environments is currently the best-supported way of [deploying different resources in different environments](https://medium.com/johnveldboom/create-terraform-resources-only-in-certain-environments-dde0f20578a5).